### PR TITLE
Allow kamon metrics to be enable/disable as per need. 

### PR DESCRIPTION
**Pull request by parekuti**

### CODE REVIEW

File: cassandra/src/main/scala/filodb.cassandra/columnstore/ChunkRowMapTable.scala [hunk 1]
```diff
@@ -121,16 +122,19 @@ sealed class ChunkRowMapTable(dataset: DatasetRef, connector: FiloCassandraConne
...
   def writeChunkMap(partition: Types.BinaryPartition,
                     version: Int,
                     segmentId: Types.SegmentId,
                     chunkIds: ByteBuffer,
                     rowNums: ByteBuffer,
                     nextChunkId: Int,
-                    stats: ColumnStoreStats): Future[Response] = {
-    stats.addIndexWriteStats(chunkIds.capacity.toLong + rowNums.capacity.toLong + 4L)
+                    stats: Option[ColumnStoreStats]): Future[Response] = {
+    if(stats.nonEmpty) {
```
Comment (velvia) [inline cassandra/src/main/scala/filodb.cassandra/columnstore/ChunkRowMapTable.scala]: space between if and (

File: cassandra/src/main/scala/filodb.cassandra/columnstore/ChunkTable.scala [hunk 2]
```diff
@@ -62,7 +62,9 @@ sealed class ChunkTable(dataset: DatasetRef, connector: FiloCassandraConnector)
       writeChunksCql.bind(partBytes, version: java.lang.Integer, segKeyBytes,
                           id: java.lang.Integer, columnName, bytes)
     }.toSeq
-    stats.addChunkWriteStats(statements.length, chunkBytes)
+    if(stats.nonEmpty) {
```
Comment (velvia) [inline cassandra/src/main/scala/filodb.cassandra/columnstore/ChunkTable.scala]: space

File: coordinator/src/main/scala/filodb.coordinator/CoordinatorSetup.scala [hunk 3]
```diff
@@ -84,7 +84,10 @@ trait CoordinatorSetup {
     metaStore.shutdown()
     // Important: shut down executioncontext as well
     threadPool.shutdown()
-    kamon.Kamon.shutdown()
+    val kamonEnabled = config.getBoolean("kamon-metrics-flag-enabled")
+    if(kamonEnabled) {
```
Comment (velvia) [inline coordinator/src/main/scala/filodb.coordinator/CoordinatorSetup.scala]: space

File: coordinator/src/main/scala/filodb.coordinator/DatasetCoordinatorActor.scala [hunk 4]
```diff
@@ -138,16 +138,43 @@ private[filodb] class DatasetCoordinatorActor(projection: RichProjection,
...
 
   // ==== Kamon metrics ====
-  private val kamonTags = Map("dataset" -> datasetName, "version" -> version.toString)
-  val kamonRowsIngested = Kamon.metrics.counter("dca-rows-ingested", kamonTags)
-  val kamonActiveRows   = Kamon.metrics.gauge("dca-active-rows", kamonTags) { activeRows.toLong }
-  val kamonFlushingRows = Kamon.metrics.gauge("dca-flushing-rows", kamonTags) {
-    Math.max(flushingRows, 0).toLong
+  val kamonEnabled = config.getBoolean("kamon-metrics-flag-enabled")
+  private val kamonTags = {
+    if(kamonEnabled) {
```
Comment (velvia) [inline coordinator/src/main/scala/filodb.coordinator/DatasetCoordinatorActor.scala]: space space space

File: coordinator/src/main/scala/filodb.coordinator/DatasetCoordinatorActor.scala [hunk 5]
```diff
@@ -194,7 +221,9 @@ private[filodb] class DatasetCoordinatorActor(projection: RichProjection,
   private def writeToMemTable(): Unit = if (tempRows.nonEmpty) {
     logger.debug(s"Flushing ${tempRows.length} rows to MemTable...")
     rowsIngested += tempRows.length
-    kamonRowsIngested.increment(tempRows.length)
+    if (kamonEnabled) {
+      kamonRowsIngested.get.increment(tempRows.length)
```
Comment (velvia) [inline coordinator/src/main/scala/filodb.coordinator/DatasetCoordinatorActor.scala]: Just do `kamonRowsIngested.foreach(_.increment(...))`

File: core/src/main/scala/filodb.core/store/ColumnStore.scala [hunk 6]
```diff
@@ -214,23 +218,27 @@ trait CachedMergingColumnStore extends ColumnStore with ColumnStoreScanner with
...
 
-  def appendSegment(projection: RichProjection,
-                    segment: Segment,
-                    version: Int): Future[Response] = Tracer.withNewContext("append-segment") {
+  private def appendSegmentWithTrace(projection: RichProjection,
+                                     segment: Segment,
+                                     version: Int): Future[Response] = Tracer.withNewContext("append-segment") {
     val ctx = Tracer.currentContext
-    stats.segmentAppend()
+    if(stats.nonEmpty) {
```
Comment (velvia) [inline core/src/main/scala/filodb.core/store/ColumnStore.scala]: space between if and (

**Comment by velvia** [comment]
@parekuti oh, can you also remove all of the .idea file changes from the PR?  Thanks.

**Comment by parekuti** [comment]
@velvia Thank you for your time and comments.

1. Added extra space for if else blocks.
2. Removed .idea files
3. Yes, agree this is a temporary fix for us to get going until we fix the kamon metrics issue.
4. It's not just ColumnStoreStats that i should create kamon-less classes, there are other objects/classes that i should modify to completely get away with Option[].  Let me know if all objects required to be changed.

File: core/src/main/scala/filodb.core/store/ColumnStore.scala [hunk 7]
```diff
@@ -271,9 +271,9 @@ trait CachedMergingColumnStore extends ColumnStore with ColumnStoreScanner with
   def appendSegment(projection: RichProjection,
                     segment: Segment,
                     version: Int): Future[Response] = {
-    if(stats.isEmpty){
+    if (stats.isEmpty){
```
Comment (velvia) [inline core/src/main/scala/filodb.core/store/ColumnStore.scala]: Also space between ) and {

**Comment by velvia** [comment]
@parekuti this is really old, and I think it's fine to squash and merge it now.

**Pull request merged**